## 18_check_snmp_process_tmp_file by S.Bay
##
## DP: Fix tmp file check_snmp_process

## From 88c3bfd626a531e7da1c3e3c694f6fd1d4a70692 Mon Sep 17 00:00:00 2001
## From: "S.Bay" <info@space2place.de>
## Date: Tue, 17 Dec 2013 09:32:15 +0100
## Subject: [PATCH] Fix error writing tmp file using plugin with remote unix
##  system

---
 check_snmp_process.pl | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/check_snmp_process.pl b/plugins/check_snmp_process.pl
index 7a83432..11f2d96 100755
--- a/check_snmp_process.pl
+++ b/check_snmp_process.pl
@@ -640,6 +640,10 @@ sub check_options {
   #### Read file
   $temp_file_name=$o_descr;
   $temp_file_name =~ s/ /_/g;
+  $temp_file_name =~ s/\//_/g;
+  $temp_file_name =~ s/-//g;
+  $temp_file_name =~ s/=//g;
+  $temp_file_name = substr($temp_file_name,0,40);
   $temp_file_name = $o_base_dir . $o_host ."." . $temp_file_name; 
   # First, read entire file
   my @ret_array=read_file($temp_file_name,$n_items_check);
-- 
1.8.5.5

